org.adempiere.base/src/org/adempiere/process/UUIDGenerator.java

```diff
@@ -129,18 +129,32 @@ protected String doIt() throws Exception {
 
 					//update db
 					if (isFillUUID) {
-						// COMMENT NEXT LINE ON RELEASE WORK
 						String msg = updateUUID(mColumn, null);
 						if (! Util.isEmpty(msg)) {
 							addBufferLog(0, null, null, msg, 0, 0);
 						}
 					}
 				} else {
-					if (isFillUUID) {
-						MColumn mColumn = MColumn.get(getCtx(), AD_Column_ID);
-						// COMMENT NEXT LINE ON RELEASE WORK
-						String msg = updateUUID(mColumn, null);
-						if (! Util.isEmpty(msg)) {
+					MColumn column = MColumn.get(AD_Column_ID);
+					if (column.isActive()) {
+						if (isFillUUID) {
+							MColumn mColumn = MColumn.get(getCtx(), AD_Column_ID);
+							String msg = updateUUID(mColumn, null);
+							if (! Util.isEmpty(msg)) {
+								addBufferLog(0, null, null, msg, 0, 0);
+							}
+						}
+					} else {
+						StringBuilder sqlclear = new StringBuilder("UPDATE ")
+								.append(cTableName)
+								.append(" SET ")
+								.append(columnName)
+								.append("=NULL WHERE ")
+								.append(columnName)
+								.append(" IS NOT NULL");
+						int cnt = DB.executeUpdateEx(sqlclear.toString(), get_TrxName());
```

Comment:
this auto clearing could be dangerous as it didn't warn user before hand.

Reply:
Thanks @hengsin - I fixed that adding a new parameter (IsClearUUID default to false)

During the tests I noticed the buffer log was not being flushed, that's a problem for any process that uses local transactions, so I added a minor fix for that.